crates/timers/src/callback.rs

```diff
@@ -154,7 +154,7 @@ impl Interval {
     {
         let mut callback = Some(callback);
         let closure = Closure::wrap(Box::new(move || {
-            let mut callback = callback.take().unwrap_throw();
+            let callback = callback.as_mut().take().unwrap_throw();
```

Comment:
I think this could be simplified to:
```
let closure = Closure::wrap(callback);`
```

Reply:
Because of `take()`, I'm pretty sure this will still break.

So I agree with @OddCoincidence, that this should just completely remove the `Some` wrapper and use `Closure::wrap(Box::new(callback) as Box<FnMut()>)` instead

Reply:
Yeah I already updated my local copy to `Closure::wrap(Box::new(callback) as Box<FnMut()>)`, I'm just grappling with the integration tests at the moment and should have the changes pushed up soon.

Reply:
Also the current version with `.take()` does work and is re-entrant. It works because `as_mut` is creating a new Option that is a mutable reference. So `take()` on a `.as_mut()` is not consuming the original Option.